### PR TITLE
Added Button Base Component

### LINKED ISSUE #279: Buttons
see #260

### PR DESCRIPTION
**Pull request by ClientCrash**
**Description**
Added Button Base Component

- [x] Add Button Component
- [x] Add Styling
- [x] Create proper onClick  

**Issue**
Closes #279

### CODE REVIEW

File: src/app/design/styled-button/styled-button.component.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+})
+export class StyledButtonComponent implements OnInit {
+
+  @Input() disabled = false;
+  @Input() styleClass: string;
+  ngOnInit() {
+
+  }
+
+  @Input() onClick: (event: Event) => void = () => {};
```
Comment (MarcelCoding) [inline src/app/design/styled-button/styled-button.component.ts:15]: If you want to pass data to a parent use a [EventEmitter](https://angular.io/api/core/EventEmitter#usage-notes).

I would do it: `<app-styled-button (click)="...">`. @MaxiHuHe04 is there a problem if the click event is not on the button but on the component?

File: src/app/design/styled-button/styled-button.component.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  templateUrl: './styled-button.component.html',
+  styleUrls: ['./styled-button.component.scss']
+})
+export class StyledButtonComponent implements OnInit {
+
+  @Input() disabled = false;
+  @Input() styleClass: string;
+  ngOnInit() {
+
+  }
```
Comment (MarcelCoding) [inline src/app/design/styled-button/styled-button.component.ts:13]: Remove empty methods.

File: src/app/design/styled-button/styled-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Component, Input, OnInit} from '@angular/core';
+@Component({
+  selector: 'app-styled-button',
+  templateUrl: './styled-button.component.html',
+  styleUrls: ['./styled-button.component.scss']
+})
+export class StyledButtonComponent implements OnInit {
+
+  @Input() disabled = false;
+  @Input() styleClass: string;
```
Comment (MarcelCoding) [inline src/app/design/styled-button/styled-button.component.ts:10]:
```suggestion
  @Input() styleClass: 'primary' | 'success' | 'warning' | 'danger' | 'info' = 'primary';
```
Also, name it some the variable something meaningful e.g. `type`.

Maybe use some namingfull names for the types? @MaxiHuHe04

File: src/app/design/styled-button/styled-button.component.scss [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  color: #2F2F36;
+  background: $info-default;
+  	&:disabled{
+      background-color: $info-disabled;
+    }
+    &:hover{
+      background-color: $info-hover;
+      color: white;
+    }
+}
```
Comment (MarcelCoding) [inline src/app/design/styled-button/styled-button.component.scss:57]: Please format the CSS with your editor.

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

**Comment by ClientCrash** [comment]
Todo: Make new click method.

File: src/app/design/styled-button/styled-button.component.scss [hunk 5]
```diff
@@ -0,0 +1,71 @@
+@import '../theme-colors';
```
Comment (MarcelCoding) [inline src/app/design/styled-button/styled-button.component.scss:1]: Use global scss (see `_variables.scss`)

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

File: src/app/design/styled-button/styled-button.component.scss (hunk 5 above)
Comment (ClientCrash) [reply src/app/design/styled-button/styled-button.component.scss:1]: I used the theme colors file to be able to add more themes later (if wanted).

Comment (ClientCrash) [reply src/app/design/styled-button/styled-button.component.scss:1]: Its also easier to edit the theme data if its in a specific file.

**Comment by MarcelCoding** [comment]
Then you should use CSS variables because you can change them at runtime scss variables are fixed at build time.

File: src/app/design/styled-button/styled-button.component.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  selector: 'app-styled-button',
+  templateUrl: './styled-button.component.html',
+  styleUrls: ['./styled-button.component.scss']
+})
+export class StyledButtonComponent {
+
+  @Input() disabled = false;
+
+
+  @Input() styleClass: 'primary' | 'success' | 'warning' | 'danger' | 'info' = 'primary';
```
[the remark this answers is not in the document]
Comment (Tristan-H11) [reply src/app/design/styled-button/styled-button.component.ts:12]: @MarcelCoding please check

File: src/global-styles/_theme-colors.scss [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+$danger-hover: #B0002F;
+$danger-disabled: #671E35;
+/*info*/
+$info-default: #3281F6;
+$info-hover: #245EB5;
+$info-disabled: #2D456C;
+/*etc*/
+$logo-crypt: #27A036
+/* ISSUE : #260 */
+
```
[the remark this answers is not in the document]
Comment (Tristan-H11) [reply src/global-styles/_theme-colors.scss:24]: @MarcelCoding https://github.com/cryptic-game/frontend/pull/272 is approved an looks ready to merge

File: src/app/design/styled-button/styled-button.component.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (ClientCrash) [reply src/app/design/styled-button/styled-button.component.ts]: Done

File: ComponentDocs.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+- info
+
+---
+
+## Button
+
+### Example
+
+```html
+ <app-styled-button (click)="login()" styleClass="primary" [disabled]="!form.valid">Log in</app-styled-button>
```
Comment (MarcelCoding) [inline ComponentDocs.md:18]:
```suggestion
 <app-styled-button (click)="login()" flavor="primary" [disabled]="!form.valid">Log in</app-styled-button>
```

File: src/app/design/styled-button/styled-button.component.scss [hunk 9]
```diff
@@ -0,0 +1,79 @@
+@import 'theme-colors';
+button{
```
Comment (MarcelCoding) [inline src/app/design/styled-button/styled-button.component.scss:2]: Format CSS, maybe use scss loop and map.

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

**Review by MarcelCoding — CHANGES_REQUESTED** [review]
Besides the ugly formatted CSS, it seems like everything is fine now.

**Review by MarcelCoding — APPROVED** [review]

File: src/app/design/styled-button/styled-button.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { StyledButtonComponent } from './styled-button.component';
+
+describe('StyledButtonComponent', () => {
```
Comment (Tristan-H11) [inline src/app/design/styled-button/styled-button.component.spec.ts:5]: With version 3, we want to encourage the use of unit tests. Some basic tests for the flavors and the "disabled"-property would be nice and won't take much time! :)

**Review by Tristan-H11 — CHANGES_REQUESTED** [review]

File: src/app/design/styled-button/styled-button.component.spec.ts (hunk 10 above)
Comment (Tristan-H11) [reply src/app/design/styled-button/styled-button.component.spec.ts:5]: Everything else is fine

**Review by MarcelCoding — APPROVED** [review]

**Pull request merged**